demo/ControlPanel.html

```diff
@@ -0,0 +1,203 @@
+  }
+
+  prototype.onThemeChange = function(event) {
+    const themeClass = pagelib.ThemeTransform.THEME[event.target.value]
+    pagelib.ThemeTransform.setTheme(document, themeClass)
+
+    this.dispatchEvent(new CustomEvent('change:theme', { detail: event.target.value }))
+  }
+
+  prototype.onDimImagesChange = function(event) {
+    pagelib.DimImagesTransform.dim(window, event.target.checked)
+
+    this.dispatchEvent(new CustomEvent('change:dim_images', { detail: event.target.checked }))
+  }
+
+  prototype.onCompatibilityFilterChange = function(event) {
+    const html = document.querySelector('html')
+    const compatibilityClass = pagelib.CompatibilityTransform.COMPATIBILITY[event.target.value]
+    html.classList[event.target.checked ? 'add' : 'remove'](compatibilityClass)
+
+    this.dispatchEvent(new CustomEvent('change:compatibility_filter'),
+      { detail: event.target.checked })
+  }
+
+  prototype.onPageProtectionChange = function(event) {
+    const protectionClass = pagelib.EditTransform.CLASS.PROTECTION[event.target.value]
+    const html = document.querySelector('html')
+    if (protectionClass) {
+      html.classList.add(protectionClass)
+    }
+
+    Object.values(pagelib.EditTransform.CLASS.PROTECTION)
+      .filter(previousClass => previousClass && previousClass !== protectionClass)
+      .forEach(previousClass => html.classList.remove(previousClass))
+
+    this.dispatchEvent(new CustomEvent('change:page_protection'), { detail: event.target.value })
+  }
+
+  prototype.createdCallback = function() {
+    // eslint-disable-next-line require-jsdoc
+    const initCallback = (clone, selector, callback) => {
+      const element = clone.querySelector(selector)
+      element.addEventListener('change', callback.bind(this))
+
+      // Update the URL on change.
+      element.addEventListener('change', () => element.form.submit())
+
+      const value = queryStringValueByName(element.name)
+      const option = element.value === value ? element : element.querySelector(`[value='${value}']`)
+      if (option) {
+        option.selected = true
+        option.checked = true
+      }
+
+      // Trigger the initial state.
+      element.dispatchEvent(new Event('change'))
+    }
+
+    const template = currentDocument.querySelector('template')
+    const clone = document.importNode(template.content, true)
+
+    initCallback(clone, '.platform', this.onPlatformChange)
+    initCallback(clone, '.layout_direction', this.onLayoutDirectionChange)
+    initCallback(clone, '.theme', this.onThemeChange)
+    initCallback(clone, '#dim_images', this.onDimImagesChange)
+    initCallback(clone, '#compatibility_filter', this.onCompatibilityFilterChange)
+    initCallback(clone, '.page_protection', this.onPageProtectionChange)
+
+    const root = this.createShadowRoot()
+    root.appendChild(clone)
+  }
+
+  document.registerElement('control-panel', { prototype })</script>
```

Comment:
Looks like `registerElement` is what's hanging... looks like [it's deprecated](https://developer.mozilla.org/en-US/docs/Web/API/Document/registerElement)?